tests/unit/test_node.py

```diff
@@ -274,29 +275,31 @@ def test_get_state(self):
         Node class.
 
         """
-        node = Node(None, None)  # Head node
+        node = TransformationProxy(Node(None, None))  # Head node
         n1 = node.Define("a", b="c")  # First child node
 
         # Required dictionaries
-        node_dict = {"children": [n1]}
+        node_dict = {"children": [n1.proxied_node]}
         n1_dict = {
             'operation_name': "Define",
             'operation_args': ("a",),
             'operation_kwargs': {"b": "c"},
             'children': []
         }
-
+        # node is of class Node, its state dictionary can be directly accessed.
         self.assertDictEqual(node.__getstate__(), node_dict)
-        self.assertDictEqual(n1.__getstate__(), n1_dict)
+        # n1 is of class TransformationProxy, so the proxied node must be
+        # accessed in order to extract its dictionary.
+        self.assertDictEqual(n1.proxied_node.__getstate__(), n1_dict)
 
     def test_set_state(self):
         """
         Test cases to check the working of
         __setstate__ method on Node class.
 
         """
-        node = Node(None, None)  # Head node
-        n1 = Node(None, None)  # First child node
+        node = TransformationProxy(Node(None, None))  # Head node
```

Comment:
In lines 313 and 314, we need to call the `__set_state__` method of the `proxied_node`.

Reply:
done